### PR TITLE
Qnumber support

### PR DESCRIPTION
**Pull request by southeo**
Resolve qnumbers for specimenHosts

### CODE REVIEW

File: src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java [hunk 1]
```diff
@@ -585,18 +587,25 @@ private List<HandleAttribute> setSpecimenHostName(DigitalSpecimenRequest request
               SPECIMEN_HOST_NAME.get(),
               specimenHostName.getBytes(StandardCharsets.UTF_8)));
     } else {
+      String specimenHostNameResolved;
       try {
-        var specimenHostNameResolved = pidResolver.getObjectName(getRor(request.getSpecimenHost()));
+        String specimenHostId = request.getSpecimenHost();
+        if(specimenHostId.contains(ROR_DOMAIN)){
+          specimenHostNameResolved = pidResolver.getObjectName(getRor(specimenHostId));
+        } else{
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java:595]: Maybe also put an else if here for the WikiData and throw an error when anything else.

File: src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java [hunk 2]
```diff
@@ -585,18 +587,25 @@ private List<HandleAttribute> setSpecimenHostName(DigitalSpecimenRequest request
...
         fdoRecord.add(
             new HandleAttribute(SPECIMEN_HOST_NAME.index(), handle,
                 SPECIMEN_HOST_NAME.get(),
                 specimenHostNameResolved.getBytes(StandardCharsets.UTF_8)));
-
-      } catch (Exception e) {
+      } catch (NotFoundException | UnprocessableEntityException | InvalidRequestException |
+               PidResolutionException e ) {
         log.error(
             "SpecimenHostId is not a resolvable ROR and no SpecimenHostName is provided in the request. SpecimenHostName field left blank. More information: "
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java:605]: Do we need to mention WikiData in the log message?

**Review by samleeflang — COMMENTED** [review]
Nice, only a small comment

File: src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java (hunk 2 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/service/FdoRecordService.java:605]: Added log message to the webclient class (pid resolver).

**Review by samleeflang — APPROVED** [review]

**Pull request merged**